src/data/finset/noncomm_prod.lean

```diff
@@ -217,6 +217,42 @@ by simp [noncomm_prod, insert_val_of_not_mem ha, multiset.noncomm_prod_cons']
     (λ x hx y hy, by rw [mem_singleton.mp hx, mem_singleton.mp hy]) = f a :=
 by simp [noncomm_prod, multiset.singleton_eq_cons]
 
+@[to_additive]
+lemma noncomm_prod_map {α : Type*} {β : Type*} [monoid β]
+  (s : finset α) (f : α → β)
```

Comment:
I think there is already an `α` and `β` of the right type floating around
```suggestion
lemma noncomm_prod_map (s : finset α) (f : α → β)
```
It would probably make sense to rename `M` to `γ` and put it right at the top of the file too.